### PR TITLE
Fix typed cluster singleton cross dc proxies

### PR DESCRIPTION
**Pull request by chbatey**
- Adds first multi-jvm test for typed cluster
- New API on ClusterSingleton: crossDcProxy

Refs #24935

### CODE REVIEW

**Comment by chbatey** [comment]
Also adding a typed multi dc sharding test here

File: akka-cluster-typed/src/main/scala/akka/cluster/typed/ClusterSingleton.scala [hunk 1]
```diff
@@ -142,6 +146,13 @@ abstract class ClusterSingleton extends Extension {
     terminationMessage: A
   ): ActorRef[A]
 
+  /**
+   * Create, if needed, a proxy to a singleton in another data center. Will not attempt to create a Singleton Manager.
+   *
+   * @param singletonName Name of the singleton
+   * @return A proxy actor that can be used to communicate with the singleton in the cluster
+   */
+  def crossDcProxy[A](singletonName: String, settings: ClusterSingletonSettings): ActorRef[A]
```
Comment (johanandren) [inline akka-cluster-typed/src/main/scala/akka/cluster/typed/ClusterSingleton.scala]: Hmm, shouldn't we rather have an overload with a DC param for the other one and keep the API nicely single-method:y?

**Review by johanandren — COMMENTED** [review]
Code looks alright but question about the API

File: akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/typed/MultiDcClusterShardingSpec.scala [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package akka.cluster.sharding.typed
```
Comment (ktoso) [inline akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/typed/MultiDcClusterShardingSpec.scala]: header? does not add here automatically? consider fixing the plugin config to do so

File: akka-cluster-typed/src/main/scala/akka/cluster/typed/ClusterSingleton.scala (hunk 1 above)
Comment (chbatey) [reply akka-cluster-typed/src/main/scala/akka/cluster/typed/ClusterSingleton.scala]: I find it odd I need to provide the behavior and shutdown message to just get a proxy. What do others think /cc @akka/akka-team

File: akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/typed/MultiDcClusterShardingSpec.scala (hunk 2 above)
Comment (chbatey) [reply akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/typed/MultiDcClusterShardingSpec.scala]: I fixed the plugin but it is a lot of changes. I'll push that as a diff PR to keep this one clean.

**Comment by chbatey** [comment]
@akka/akka-team this is ready for a re-review. I've removed the API change and now it is just the bug fix + additional testing.

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala [hunk 3]
```diff
@@ -678,8 +675,8 @@ private[akka] class ShardRegion(
     coordinatorSelection.foreach(_ ! registrationMessage)
     if (shardBuffers.nonEmpty && retryCount >= 5)
       log.warning(
-        "Trying to register to coordinator at [{}], but no acknowledgement. Total [{}] buffered messages.",
-        coordinatorSelection, shardBuffers.totalSize)
+        "Trying to register to coordinator at [{}], but no acknowledgement. Total [{}] buffered messages. Members [{}]",
+        coordinatorSelection, shardBuffers.totalSize, membersByAge)
```
Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: There is another open PR https://github.com/akka/akka/pull/24906 that is improving this log message, so perhaps revert this change. I'm not sure we should log all members here.

**Review by patriknw — APPROVED** [review]
LGTM after reverting one log message

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala (hunk 3 above)
Comment (chbatey) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: k, will remove
